src/vs/workbench/services/files/node/fileService.ts

```diff
@@ -195,6 +196,111 @@ export class FileService implements IFileService {
 		return this.doResolveContent(resource, options, (stat, enc) => this.resolveFileContent(stat, enc));
 	}
 
+	private static chunkSize = 64 * 1024;
+	private static chunkBuffer = Buffer.allocUnsafe(FileService.chunkSize);
+
+	public resolveStringStream(resource: uri, options?: IResolveContentOptions, token: CancellationToken = CancellationToken.None): IStringStream {
+		const ret = new EventEmitter();
+
+		fs.open(resource.path, 'r', (err, fd) => {
+
+			if (err) {
+				ret.emit('error', err);
+				return;
+			}
+
+			let decoder: NodeJS.ReadWriteStream;
+			let totalBytesRead = 0;
+
+			const finish = (err?: any) => {
+				if (err) {
+					ret.emit('error', err);
+				}
+				fs.close(fd, err => {
+					if (err) {
+						ret.emit('error', err);
+					}
+					if (decoder) {
+						decoder.removeAllListeners();
+					}
+					ret.emit('end');
+				});
+			};
+
+			const handleChunk = (bytesRead) => {
```

Comment:
Put this logic to private separate method of this class.
Remove comments in conditions.

```typescript
private someAwesomeNameForMethod (bytesRead: type): void {
			if (token.isCancellationRequested) {
 					finish(new Error('cancelled'));
				} else if (bytesRead < FileService.chunkSize) {
					decoder.write(FileService.chunkBuffer.slice(0, bytesRead), finish);
				} else {
					decoder.write(FileService.chunkBuffer, readChunk);
				}
	}
```

Reply:
Then you can reuse this. And you don't need assign it to constant